### ISSUE TITLE
Predictions: Should predict a missing class (Predict on test data)

### ISSUE THREAD

**Issue by Pelonza** [biolab/orange3#2148]
##### Orange version

3.4.*

##### Expected behavior

For most machine learning applications, you want to be able to use a trained predictor/classifier or other algorithm to NEW data, which doesn't have an outcome already determined.

##### Actual behavior

Right now, to use the "predict" widget, the documentation recommends taking a random sample of the input data, then training the models on one sub-set, and predicting the other set.

##### Steps to reproduce the behavior

Follow your example model to produce the regular behavior...
I was trying to do the Titanic Kaggle.com challenge, where you build a classifier on training data, then predict survive/die on NEW test data... this doesn't work with importing Kaggle.com's base data-sets. I've been struggling to find a way to make it work at all....

##### Additional info (worksheets, data, screenshots, ...)

I think the key here would be to add functionality to the "predict" module that detects if there is new data missing the target column (as defined in the trained classifiers/models). Then, if that column is missing, attach a "blank" column with the same attributes and populate it with the outcome from the trained input models (similar to what's going on already with data that already HAS a target column, but being a bit smarter about detecting it missing).

**Comment by janezd** [comment]
We thought we fixed this bug already. This is critical.

**Comment by jerneju** [comment]
It works in 3.3.12 but not in 3.4.0.

**Comment by janezd** [comment]
Is this related to merging of classifiers and regressors? @pavlin-policar?

**Comment by pavlin-policar** [comment]
I really don't think so... My changes don't touch the models at all, or in how they are created, which is where this seems to fail. My changes don't even really touch the "learning" process much.

**Comment by pavlin-policar** [comment]
I've isolated the the issue to a commit, `e977e8b` (#2026). The issue here, this should be helpful.

I've reproduced the error by splitting a dataset, then removing the target class column from the test set, then sending a trained model to and the test data to the Predictions widget.

**Comment by janezd** [comment]
Ooops, that was my fault! 😟 

Thanks, @pavlin-policar. Now I have a task for Friday.

**Comment by Pelonza** [comment]
@pavlin-policar I wonder if this is al
